### DISCUSSION TITLE
RESTful Endpoint Generation

### DISCUSSION THREAD

**Discussion by Brian-McBride** [trpc/trpc#1878] [Ideas]
## RESTful (OpenAPI) export

After looking through the docs, I am liking what **trpc** is doing here. Like others have probably commented elsewhere - we don't need yet another communication spec. However, leveraging `JSON-RPC 2.0 Specification` provides some capability with other libs and other languages. 

Sadly, I still have clients that are asking for RESTful endpoints. It occurred to me that it wouldn't be a far leap to also dynamically generate an endpoint that follows along with the OpenAPI spec. I'm curious if anyone knows if this has been done - or if there are others with interest in some middleware that can generate APIs that follow the RPC pattern. 

Obviously, it is simple to just add your own route to the base server (express, fastify, etc...) and use a more traditional handler instead of trpc. Stil, since trpc has a schema, auto generating these endpoints and possibly feed a swagger portal could be pretty cool for situations where some of your APIs are public and you want to go with RESTful on them while still reducing human error through automating these endpionts as well.

**Comment by KATT** [comment]
@jlalmes just added something here https://trpc.io/docs/awesome-trpc

**Reply by jlalmes** [reply]
Checkout [`trpc-openapi`](https://github.com/jlalmes/trpc-openapi) to add REST endpoints for your tRPC procedures.
